.pipelines/templates/deploy-stage.yml

```diff
@@ -1,130 +1,170 @@
+      steps:
+          - task: Bash@3
+            displayName: 'Wait and Smoke test'
+            timeoutInMinutes: 10
+            inputs:
+                targetType: 'inline'
+                script: |
+                    set -ux  # do not fail on error
 
-  - job: SwapBothSlots
-    condition: eq(dependencies.TestDeployment.result, 'Succeeded')
-    strategy:
-      matrix:
-        Analyzer:
-          Name: 'Analyzer'
-          AppServiceName: ${{ parameters.ANALYZER_APP_NAME }}
-        Anonymizer:
-          Name: 'Anonymizer'
-          AppServiceName: ${{ parameters.ANONYMIZER_APP_NAME }}
-    displayName: Swap slots from staging to production
-    dependsOn: TestDeployment
-    steps:
-      - checkout: none
-      - task: AzureAppServiceManage@0
-        displayName: Swap $(Name) slot
-        inputs:
-          azureSubscription: ${{ parameters.AZURE_SUBSCRIPTION }}
-          action: Swap Slots
-          webAppName: $(AppServiceName)
-          sourceSlot: staging
-          resourceGroupName: ${{ parameters.RESOURCE_GROUP_NAME }}
\ No newline at end of file
+                    ANALYZER_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $ANALYZER_STAGING_URI/health)
+                    ANONYMIZER_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $ANONYMIZER_STAGING_URI/health)
+                    IMAGE_REDACROE_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $IMAGE_REDACROE_STAGING_URI/health)
+                    echo analyzer status code is $ANALYZER_STATUS_CODE
+                    echo anonymizer status code is $ANONYMIZER_STATUS_CODE
+
+                    while [ ! $ANALYZER_STATUS_CODE == 200 ] || [ ! $ANONYMIZER_STATUS_CODE == 200 ] || [! $IMAGE_REDACROE_STATUS_CODE == 200]
+                    do
+                        sleep 30
+                        ANALYZER_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $ANALYZER_STAGING_URI/health)
+                        ANONYMIZER_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $ANONYMIZER_STAGING_URI/health)
+                        IMAGE_REDACROE_STATUS_CODE=$(curl -s -o /dev/null -w "%{http_code}" --connect-timeout 1 --max-time 10 $IMAGE_REDACROE_STAGING_URI/health)
+                        echo analyzer status code is $ANALYZER_STATUS_CODE
+                        echo anonymizer status code is $ANONYMIZER_STATUS_CODE
+                        echo image-redactor status code is $IMAGE_REDACROE_STATUS_CODE
+                    done
+          - template: ./e2e-tests.yml
+            parameters:
+                test_suite: 'integration'
+                analyzer_base_url: $(ANALYZER_STAGING_URI)
+                anonymizer_base_url: $(ANONYMIZER_STAGING_URI)
+                image_redactor_base_url: $(IMAGE_REDACROE_STAGING_URI)
```

Comment:
image_redactor_base_url  Needs to be added to e2e-tests.yml and e2e framework, can be done when adding e2e tests

Reply:
I'm going to remove the ci pipeline change and do it in a seperate PR

Reply:
Just what I wanted to suggest, remember that yesterday it was good for merge